wp-includes/rest-api/endpoints/class-wp-rest-customize-changesets-controller.php

```diff
@@ -560,6 +561,157 @@ public function create_item( $request ) {
+		}
+
+		// TODO (?): Not filterable a la WP_REST_Posts_Controller.
+		if ( ! ( EMPTY_TRASH_DAYS > 0 ) ) {
+			return new WP_Error(
+				'rest_trash_not_supported',
+				__( 'The post does not support trashing. Set force=true to delete.' ),
+				array(
+					'status' => 501,
+				)
+			);
+		}
+
+		if ( 'trash' === get_post_status( $manager->changeset_post_id() ) ) {
+			return new WP_Error(
+				'rest_already_trashed',
+				__( 'The changeset has already been deleted.' ),
+				array(
+					'status' => 410,
+				)
+			);
+		}
+
+		// TODO: Ripped straight from _wp_customize_publish_changeset().
+
+		/** This action is documented in wp-includes/post.php */
+		do_action( 'wp_trash_post', $manager->changeset_post_id() );
+
+		add_post_meta( $manager->changeset_post_id(), '_wp_trash_meta_status', $post->post_status );
+		add_post_meta( $manager->changeset_post_id(), '_wp_trash_meta_time', time() );
+
+		$old_status = $post->post_status;
+		$new_status = 'trash';
+		$wpdb->update( $wpdb->posts, array( 'post_status' => $new_status ), array( 'ID' => $manager->changeset_post_id() ) );
+		clean_post_cache( $manager->changeset_post_id() );
+
+		$post->post_status = $new_status;
+		wp_transition_post_status( $new_status, $old_status, $post );
+
+		/** This action is documented in wp-includes/post.php */
+		do_action( 'edit_post', $manager->changeset_post_id(), $post );
+
+		/** This action is documented in wp-includes/post.php */
+		do_action( "save_post_{$post->post_type}", $manager->changeset_post_id(), $post, true );
+
+		/** This action is documented in wp-includes/post.php */
+		do_action( 'save_post', $manager->changeset_post_id(), $post, true );
+
+		/** This action is documented in wp-includes/post.php */
+		do_action( 'wp_insert_post', $manager->changeset_post_id(), $post, true );
+
+		/** This action is documented in wp-includes/post.php */
+		do_action( 'trashed_post', $manager->changeset_post_id() );
+
+		// TODO: At this point $wp_customize will no longer have up-to-date post data.
+		$result = get_post( $manager->changeset_post_id() );
+
+		if ( ! $result ) {
+			return new WP_Error(
+				'rest_cannot_delete',
+				__( 'The post cannot be deleted.' ),
+				array(
+					'status' => 500,
+				)
+			);
+		}
+
+		return $this->prepare_item_for_response( $result, $request );
+	}
+
+	/**
+	 * Check whether a request can delete the specified changeset.
+	 *
+	 * @since ?.?.?
+	 * @access public
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return bool|WP_Error True if the request has access to delete the item, otherwise false or WP_Error object.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		$manager = $this->ensure_customize_manager( $request['uuid'] );
+
+		if ( ! $manager->changeset_post_id() ) {
+			return new WP_Error(
+				'rest_post_invalid_uuid',
+				__( 'Invalid changeset UUID.' ),
+				array(
+					'status' => 404,
+				)
+			);
+		}
+
+		// TODO: Use 'delete_post'.
```

Comment:
Probably a new Trac ticket.

Some prior discussions on that: 
* https://wordpress.slack.com/archives/C0381N237/p1481851138000475
* https://wordpress.slack.com/archives/D0G14JH09/p1489768662369477 (our DM convo)
* https://github.com/xwp/wp-customize-snapshots/pull/4
* https://github.com/xwp/wp-customize-snapshots/pull/74
* https://core.trac.wordpress.org/ticket/38900#comment:7